tools/sdk-generation-pipeline/packages/sdk-generation-lib/src/lib/ArtifactUploader.ts

```diff
@@ -0,0 +1,189 @@
+    child_process.execSync('git add .', {
+        encoding: 'utf8',
+        cwd: packageFolder,
+    });
+    const files = child_process.execSync('git ls-files', { encoding: 'utf8', cwd: packageFolder }).trim().split('\n');
+
+    return files;
+}
+
+export type ArtifactBlobUploaderContext = {
+    azureStorageBlobSasUrl: string;
+    azureBlobContainerName: string;
+    language: string;
+    pipelineBuildId: string;
+};
+
+export class ArtifactBlobUploader {
+    private azureBlobClient: AzureBlobClient;
+    private language: string;
+    private pipelineBuildId: string;
+
+    constructor(artifactBlobUploaderContext: ArtifactBlobUploaderContext) {
+        this.validateContext(artifactBlobUploaderContext);
+        this.azureBlobClient = new AzureBlobClient(
+            artifactBlobUploaderContext.azureStorageBlobSasUrl,
+            artifactBlobUploaderContext.azureBlobContainerName
+        );
+        this.language = artifactBlobUploaderContext.language;
+        this.pipelineBuildId = artifactBlobUploaderContext.pipelineBuildId;
+    }
+
+    private validateContext(artifactBlobUploaderContext: ArtifactBlobUploaderContext) {
+        if (!artifactBlobUploaderContext.azureStorageBlobSasUrl) {
+            throw new Error(`Invalid azureStorageBlobSasUrl`);
+        }
+        if (!artifactBlobUploaderContext.azureBlobContainerName) {
+            throw new Error(`Invalid azureBlobContainerName`);
+        }
+        if (!(<any>Object).values(SDK).includes(artifactBlobUploaderContext.language)) {
+            throw new Error(`Invalid language`);
+        }
+        if (!artifactBlobUploaderContext.pipelineBuildId) {
+            throw new Error(`Invalid pipelineBuildId`);
+        }
+    }
+
+    public async uploadSourceCode(generateAndBuildOutputJson: GenerateAndBuildOutput) {
+        for (const p of generateAndBuildOutputJson.packages) {
+            const result = p.result;
+            if (result === 'failed') {
+                logger.warn(`Build ${p.packageName} failed, skipped it`);
+                continue;
+            }
+            const packageName = p.packageName;
+            const packageFolder = p.packageFolder;
+
+            if (packageFolder && fs.existsSync(packageFolder)) {
+                for (const filePath of getFileListInPackageFolder(packageFolder)) {
+                    if (fs.existsSync(path.join(packageFolder, filePath))) {
+                        await this.azureBlobClient.publishBlob(
+                            path.join(packageFolder, filePath),
+                            `${this.pipelineBuildId}/${this.language}/${packageName}/${filePath}`
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    public async uploadArtifacts(generateAndBuildOutputJson: GenerateAndBuildOutput) {
+        for (const p of generateAndBuildOutputJson.packages) {
+            const result = p.result;
+            if (result === 'failed') {
```

Comment:
avoid using some kind of hard-code . We have defined the type for result. use that.

Reply:
we had :
GenerateAndBuildOutputSchema.json define result: `"enum": ["failed", "succeeded", "warning"]` , and in GenerateAndBuildOutput.ts result is string type.
and we had taskResult.ts define: type PipelineResult = 'success' | 'failure' | 'time_out';
and scriptRunningState.ts : type scriptRunningState = "failed" | "pending" | "inProgress" | "succeeded" | "warning"
they are unmatched.
As we published GenerateAndBuildOutput.ts and GenerateAndBuildOutputSchema.json for all sdks, I think we shouldn't change it's value. So either we defined a new type including "failed", "succeeded", "warning" or hard-code.

Reply:
We need consistent all those definition.

Reply:
taskResult = "success" | "failure"

Reply:
done